### PR TITLE
Fetch 0x limit orders liquidity

### PR DESCRIPTION
**Pull request by squadgazzz**
# Description
Adds 0x liquidity fetcher

# Changes
- New config for the fetcher
- Requires changes in `infra` to provide api-key

## How to test
TBD

## Related Issues

Fixes https://github.com/cowprotocol/services/issues/1666

### CODE REVIEW

File: crates/driver/src/infra/liquidity/config.rs [hunk 1]
```diff
@@ -27,6 +28,9 @@ pub struct Config {
     /// The collection of Balancer V2 compatible exchanges to fetch liquidity
     /// for.
     pub balancer_v2: Vec<BalancerV2>,
+
+    /// The collection of 0x compatible exchanges to fetch liquidity for.
+    pub zeroex: Vec<ZeroEx>,
```
Comment (squadgazzz) [inline crates/driver/src/infra/liquidity/config.rs:33]: Does it make any sense to support multiple 0x exchanges?

File: crates/driver/src/infra/liquidity/config.rs [hunk 2]
```diff
@@ -231,3 +235,13 @@ impl BalancerV2 {
         })
     }
 }
+
+/// ZeroEx liquidity fetching options.
+#[derive(Clone, Derivative)]
+#[derivative(Debug)]
```
Comment (fleupold) [inline crates/driver/src/infra/liquidity/config.rs:241]: Does this crate have other advantages than ignoring a field from being printed? I think for this we could also define our own secret type which wraps string and doesn't implement Display or Debug (which might be less error prone than relying on engineers remembering to add the derivative macro)

**Review by fleupold — APPROVED** [review]
Nice, looks even easier than I thought. As to how to test, it's probably not easy to create an e2e test with this type of liquidity (we would have to mock the 0x API). Does the 0x api exist on goerli so we can craft a test trade in staging?

Otherwise we may have to merge, 👀  and 🙏

File: crates/driver/src/infra/liquidity/config.rs (hunk 2 above)
Comment (squadgazzz) [reply crates/driver/src/infra/liquidity/config.rs:241]: It is also used as follows:
1. https://github.com/cowprotocol/services/blob/main/crates/autopilot/src/database/competition.rs#L61-L62
2. https://github.com/cowprotocol/services/blob/main/crates/shared/src/zeroex_api.rs#L216-L218

> we could also define our own secret type which wraps string and doesn't implement Display or Debug

Will try this in a separate PR, but technically it's the same: don't forget to use a special type for secret fields.

Comment (fleupold) [reply crates/driver/src/infra/liquidity/config.rs:241]: > but technically it's the same: don't forget to use a special type for secret fields.

Agree, but I think it's slightly more memorable to have a deliberate secret type than remembering to use a specific derive macro, but I don't feel strongly.

**Review by sunce86 — APPROVED** [review]

**Pull request merged**